hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+          new HoodieLogFile(fsStatus[0].getPath()), writerSchema, true);
+
+      //read the avro blocks
+      while (reader.hasNext()) {
+        HoodieLogBlock n = reader.next();
+        String commitTime;
+        if (n instanceof HoodieCorruptBlock) {
+          try {
+            commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+          } catch (Exception e) {
+            commitTime = "corrupt_block_1";
+            // unable to read metadta
+          }
+        } else {
+          commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+        }
+        if (logBlockTypeToMetadata.containsKey(commitTime)) {
+          logBlockTypeToMetadata.get(commitTime).add(Pair.of(n.getBlockType(), n.getLogMetadata()));
+        } else {
+          List<Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>>> list
+              = new ArrayList<>();
+          list.add(Pair.of(n.getBlockType(), n.getLogMetadata()));
+          logBlockTypeToMetadata.put(commitTime, list);
+        }
+      }
+    }
+
+    String[][] rows = new String[logBlockTypeToMetadata.size() + 1][];
+    int i = 0;
+    for(Map.Entry<String, List<Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>>>>
+        entry : logBlockTypeToMetadata.entrySet()) {
+      for(Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>> p : entry.getValue()) {
+        String [] metadata = new String[3];
+        metadata[0] = entry.getKey();
+        metadata[1] = p.getKey().toString();
+        metadata[2] = new ObjectMapper().writeValueAsString(p.getValue());
+        rows[i] = metadata;
+        i++;
+      }
+    }
+    return HoodiePrintHelper.print(
+        new String[]{"CommitTime", "BlockType", "Metadata"}, rows);
+  }
+
+  @CliCommand(value = "show logfile records", help = "Read records from log files")
+  public String showLogFileRecords(
+      @CliOption(key = {
+          "limit"}, mandatory = false, help = "Limit commits", unspecifiedDefaultValue = "10")
+      final Integer limit,
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified paths for the log files")
+      final String logFilePathPattern,
+      @CliOption(key = "mergeRecords", mandatory = false, help = "If the records in the log files should be merged",
+          unspecifiedDefaultValue = "false")
+      final Boolean shouldMerge) throws IOException {
+
+    System.out
+        .println("===============> Showing only " + limit + " records <===============");
+
+      List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()).globStatus(new Path(logFilePathPattern)))
+          .map(status -> status.getPath().toString()).collect(Collectors.toList());
+
+    // TODO : readerSchema can change across blocks/log files, fix this inside Scanner
```

Comment:
This is handled in the RealTimeRecordReader.. is there an opp for code reuse?

Reply:
The actual problem (mentioned in TODO) is not handled yet, you've put a TODO there :)

https://github.com/uber/hudi/blob/master/hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/HoodieRealtimeRecordReader.java#L128

Apart from that, converting messagetype to avro schema is a one liner so will leave it here instead of creating a util class. When we solve the TODO, we can centralize this.